website/static/js/statistics.js

```diff
@@ -0,0 +1,213 @@
+'use strict';
+
+require('keen-dataviz/dist/keen-dataviz.min.css');
+
+var $osf = require('js/osfHelpers');
+var keenDataviz = require('keen-dataviz');
+var keenAnalysis = require('keen-analysis');
+
+var KeenViz = function(){
+    var self = this;
+
+    /**
+    * Manages the connections to Keen.  Needs a project id and valid read key for that project.
+    *
+    * @property keenClient
+    * @type {keenAnalysis}
+    */
+    self.keenClient = new keenAnalysis({
+        projectId: window.contextVars.keen.public.projectId,
+        readKey : window.contextVars.keen.public.readKey,
+    });
+
+
+    // show visits to this project over the last week
+    self.visitsByDay = function() {
+        var query = {
+            'type':'count_unique',
+            'params' : {
+                event_collection: 'pageviews',
+                timeframe: 'this_7_days',
+                interval: 'daily',
+                target_property: 'anon.id'
+            }
+        };
+
+        var dataviz = new keenDataviz()
+                .el('#visits')
+                .chartType('line')
+                .chartOptions({
+                    tooltip: {
+                        format: {
+                            name: function(){return 'Visits';}
+                        }
+                    }
+                });
+
+        self.buildChart(query, dataviz);
+    };
+
+
+    // show most common referrers to this project from the last week
+    self.topReferrers = function() {
+        var query = {
+            type: 'count_unique',
+            params: {
+                event_collection: 'pageviews',
+                timeframe: 'this_7_days',
+                target_property: 'anon.id',
+                group_by: 'referrer.info.domain'
+            }
+        };
+
+        var dataviz = new keenDataviz()
+                .el('#topReferrers')
+                .chartType('pie')
+                .chartOptions({
+                    tooltip:{
+                        format:{
+                            name: function(){return 'Visits';}
+                        }
+                    },
+                });
+
+        self.buildChart(query, dataviz);
+    };
+
+    // The number of visits by hour across last week
+    self.visitsServerTime = function() {
+        var query = {
+            'type': 'count_unique',
+            'params': {
+                event_collection: 'pageviews',
+                timeframe: 'this_7_days',
+                target_property: 'anon.id',
+                group_by: 'time.local.hour_of_day',
+            }
+        };
+
+        var dataviz = new keenDataviz()
+                .el('#serverTimeVisits')
```

Comment:
It would be good to refactor out these hardcoded CSS IDs. Instead, these should be passed at initialization time (in the page module).

Something like

```js
// statistics-page.js

projectUsageStats.
    visitsServerTime('#serverTimeVisits')
    visitsByDay('#visits')
    // ...
```